### PR TITLE
Sisu specific PreConstruct/PreDestroy annotations

### LINKED ISSUE #58: PreConstruct PreDestroy should be own annotations
Just like Description, and sisu should stop pulling in "unpopular" dependencies like javax stuff for only those two annotations. Maybe make the behave in similar way how Sisu;s `Nullable` is working (annotation from several packages are all recognized as Nullable annotation)

### PR DESCRIPTION
**Pull request by cstamas**
Just like other annotations like Typed, Priority add "own" annotations for Lifecycle as well (while do support JSR250 if present).

This closes #58

### CODE REVIEW

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2010-present Sonatype, Inc.
```
Comment (kwin) [inline org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java]: Sonatype copyright?

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Stuart McCulloch (Sonatype, Inc.) - initial API and implementation
```
Comment (kwin) [inline org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java]: Shouldn’t that be you @cstamas?

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleBuilder.java [hunk 3]
```diff
@@ -10,19 +10,37 @@
...
+import org.eclipse.sisu.PreDestroy;
 
 /**
  * Builds {@link BeanLifecycle}s by searching class hierarchies for JSR250 annotations.
  */
 final class LifecycleBuilder
 {
+    static
+    {
+        boolean hasJsr250Lifecycle;
```
Comment (kwin) [inline org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleBuilder.java]: Rather `hasJsr250Annotations` as the lifecycle concept is applicable to both types of annotations

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleModule.java [hunk 4]
```diff
@@ -23,8 +20,8 @@
 import com.google.inject.spi.TypeListener;
 
 /**
- * Guice {@link Module} that provides JSR250 lifecycle management by following {@link PostConstruct} and
- * {@link PreDestroy} annotations. The lifecycle can be controlled with the associated {@link BeanManager}.
+ * Guice {@link Module} that provides JSR250 lifecycle management by following {@link javax.annotation.PostConstruct} and
+ * {@link javax.annotation.PreDestroy} annotations. The lifecycle can be controlled with the associated {@link BeanManager}.
```
Comment (kwin) [inline org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleModule.java]: Or sisu lifecycle annotations!

**Review by kwin — CHANGES_REQUESTED** [review]

**Comment by cstamas** [comment]
Re copyright: am really unsure here, @mcculls ? The "old" files DO have (and should keep) these Sonatype headers, but am unsure whhat headers "newly added" files should carry...

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleBuilder.java (hunk 3 above)
Comment (cstamas) [reply org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleBuilder.java]: fixed

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleModule.java (hunk 4 above)
Comment (cstamas) [reply org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/bean/LifecycleModule.java]: fixed, please reread

**Comment by mcculls** [comment]
Hi @cstamas - for new files you can put your name under both the copyright and contributor lines, following the convention described in https://www.eclipse.org/projects/handbook/#ip-copyright-headers

**Comment by cstamas** [comment]
Personally, but as practical option (as linked doco lists 4th bullet as optional), I'd rather do something like "Sisu Project" or nothing at all for 4th bullet, just standard eclipse header. Wdyt?

My thinking is toward simpler formatting/validation, but may be wrong in that respect...

**Comment by mcculls** [comment]
Alternatively we could move to the more generic header and put copyright details in the `NOTICE` file - WDYT?
```
/********************************************************************************
 * Copyright (c) {year} Contributors to the Eclipse Foundation
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * SPDX-License-Identifier: EPL-1.0
 ********************************************************************************/
```

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java (hunk 1 above)
Comment (cstamas) [reply org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java]: fixed

File: org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java (hunk 2 above)
Comment (cstamas) [reply org.eclipse.sisu.inject/src/main/java/org/eclipse/sisu/PostConstruct.java]: fixed

**Comment by cstamas** [comment]
Applied header @mcculls proposed

**Review by kwin — APPROVED** [review]

**Review by mcculls — APPROVED** [review]

**Pull request merged**